src/app/vue/index.ts

```diff
@@ -26,6 +26,62 @@ const $ = {
   routesLabel: token<RouteRecordRaw[]>('vue.routes.label'),
   navigationGuards: token<NavigationGuard[]>('vue.routes.navigation.guards'),
 }
+
+// @TODO at some point we should expose this as an extendable thing
+// similar to navigationGuards. We'd then do this specific functionality in
+// `@app/application` as it relates to RouteView
+const addRouteName = (item: RouteRecordRaw, _parent?: RouteRecordRaw) => {
+  if (typeof item.component !== 'undefined' && typeof item.component === 'function') {
+    // @ts-ignore at this point item.component has to be a function because we checked it above
+    const component = item.component.bind(item)
+    item.component = async () => {
+      // we need to create a new instance of the component as import is
+      // natively cached we could re-cache this in a WeakMap if it becomes
+      // necessary (I don't think it will)
+      const cached = (await component()).default
+      const route = {
+        default: {
+          ...cached,
+          render: (...args: Record<string, unknown>[]) => {
+            args[0].$routeName = item.name
+            return cached.render(...args)
+          },
+        },
+      }
+      return route
+    }
+  }
```

Comment:
Minor thing, but I'm fan of early exits to reduce nesting 🙂

Reply:
At first I wasn't totally sure whether reversing the condtional in order to return early would make the conditional harder to follow at a glance. But I tried it out, and I simplified the conditional in doing this also 🎉 

Thanks for the suggestion! 👍 

I actually just thought I wonder if you can pass `component: import()` 🤔 👀 , gonna look it up even though we never do this.

Reply:
OK I've kept the conditional simplification, but hit some sort of issue when I started messing with static vs dynamic components (something related to the difference between dev-time and built JS), so I left the static vs dynamic to one side for now, also see other PR comments below.

Gonna resolve this convo.